### ISSUE TITLE
should all nodes be in edges (if edges exist)?

### ISSUE THREAD

**Issue by Remi-Gau** [bids-standard/stats-models#66]
All nodes mentioned in edges should refer to an existing node, but I am not sure about the other way around: have some nodes that exist but not mentioned in edges and that would thus be ignore when running the model.

Sounds practical from a user perspective to be able to develop some models where not all nodes are used while tweaking things around.

But validation could possibly throw a warning.

**Comment by effigies** [comment]
Definitely seems warn-worthy.

**Comment by Remi-Gau** [comment]
should I just add a [validator](https://pydantic-docs.helpmanual.io/usage/validators/) in the [BIDSStatsModel class](
https://github.com/bids-standard/stats-models/blob/a20a874ecdedb9f08195ec6c8cd87fb813047a36/bsmschema/bsmschema/models.py#L457) ?

**Comment by effigies** [comment]
I guess those could warn instead of raising exceptions, but if they aren't easy to skip, it will make it annoying to program with this class, as every copy will emit a new warning.

**Comment by Remi-Gau** [comment]
ah true!

so maybe something like validate method on the Edge class?

**Comment by effigies** [comment]
No, I think it still needs to operate on the whole model, as it needs to know about both nodes and edges. I can see a couple approaches. 

1. An explicit `check()` method or function that will emit warnings (could have a strict mode to make errors)
2. Have an explicitly warning subclass of `BIDSStatsModel` that implements the validation. The caller then chooses which to use for parsing.
